### DISCUSSION TITLE
Apply velocity boundary condition in an impact problem.

### DISCUSSION THREAD

**Discussion by zuckarm** [idaholab/moose#28942] [Q&A Modules: Solid mechanics]
hi all,
i wanted to know what are the different methods i can use to apply a velocity boundary condition to a particular boundary in the domain?
so far i am aware of PresetVelocity as one option.

**Comment by GiudGiud** [comment]
Hello

You can use DirichletBCs too. It depends on the rest of the simulation setup. Are you using the Dynamic Solid Mechanics Physics?

**Reply by zuckarm** [reply]
Yes i am using DynamicStressDivergenceTensors and InterialForce Kernels, along with TestNewmarkTI AuxKernels for velocity and acceleration. Can you please show an example of prescribing velocity boundary condition using FunctionDirichletBC?

**Reply by maxnezdyur** [reply]
There are multiple ways to setup the solid dynamics in MOOSE. IIRC, if you set it up similar to `solid_mechanics/test/tests/capped_weak_plane/pull_and_shear.i` then the way the `InertialForce`, velocity and acceleration is calculated you can apply a true initial condition. In this case the velocity is an `AuxVariable` that is calculated at `timestep_end` so if you set up  initial condition on the velocity and/or an auxkernel that executes on `initial` you should be able to apply this velocity IC.

**Reply by zuckarm** [reply]
hi @maxnezdyur
i looked through the input file you mentioned and i had a few questions:

1) i have a ramped velocity boundary condition which is acting throughout the simulation time. is it possible to define this ramped function and then use it in an AuxKernel defined for velocity?

2) The AuxKernels for velocity and acceleration are of the Newmark type. but i am interested in explicit time integration. what other AuxKernels can i use?

3) I am ultimately trying to solve a coupled problem: dynamic solid mechanics + phase field fracture model. so far, i have not seen any such type of problems solved in MOOSE. could you direct me to any such tutorials, if there are any?

thanks

**Reply by maxnezdyur** [reply]
Well first I misread you initial problem statement. For impact problems, I am used to IC on velocity so that's why I focused on IC so much. Let me answer your question this time!
1. If you use `PresetVelocity` BC it has a function option, so that allows you to presibe the velocity as a function from the start.
2. Since you don't need to set IC (aka they are set to 0) then don't use the auxkernels I sent you before and go back to what you first had.
3. I think the best resource for what you want is in [RACCOON](https://hugary1995.github.io/raccoon/index.html), I don't know if they have any tutorials for solid dynamics but worth a look.

**Reply by zuckarm** [reply]
thank you so much!
